book/src/proposals/validator-timestamp-oracle.md

```diff
@@ -0,0 +1,121 @@
+# Validator Timestamp Oracle
+
+Third-party users of Solana sometimes need to know the real-world time a block
+was produced, generally to meet compliance requirements for external auditors or
+law enforcement. This proposal describes a validator timestamp oracle that
+would allow a Solana cluster to satisfy this need.
+
+The general outline of the proposed implementation is as follows:
+
+- At regular intervals, each validator records its observed time for a known slot
+  on-chain (via a Timestamp added to a slot Vote)
+- A client can request a block time using the `getBlockTime` RPC method. When a
```

Comment:
This can use more definition.  It's not clear to me if this is returning the latest block's timestamp, the rooted timestamp, or an estimate of the latest timestamp using the rooted timestamp and number of slots since the latest.  And if there are forks, is that the latest latest?  If returning the rooted timestamp is sufficient, that's certainly simplest.